ui/pages/confirmation/confirmation.js

```diff
@@ -256,20 +260,37 @@ export default function ConfirmationPage({
     // viewed index, reset the index.
     if (
       pendingConfirmations.length === 0 &&
+      pendingApprovalFlows.length === 0 &&
       redirectToHomeOnZeroConfirmations
     ) {
       history.push(DEFAULT_ROUTE);
-    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+    } else if (
+      pendingConfirmations.length &&
+      pendingConfirmations.length <= currentPendingConfirmation
+    ) {
       setCurrentPendingConfirmation(pendingConfirmations.length - 1);
     }
   }, [
     pendingConfirmations,
+    pendingApprovalFlows,
     history,
     currentPendingConfirmation,
     redirectToHomeOnZeroConfirmations,
   ]);
 
+  useEffect(() => {
+    setApprovalFlowLoadingText(
+      pendingApprovalFlows.length > 0
+        ? pendingApprovalFlows[pendingApprovalFlows.length - 1].loadingText
```

Comment:
Is it worth putting this in it's own variable at the start of the method for clarity? Something like `childFlow` to be explicit where we are sourcing the loading text?

Reply:
Done.